service/awsconfig/v7/resource/lifecycle/current.go

```diff
@@ -2,9 +2,106 @@ package lifecycle
 
 import (
 	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/framework/context/resourcecanceledcontext"
+
+	cloudformationservice "github.com/giantswarm/aws-operator/service/awsconfig/v7/cloudformation"
+	"github.com/giantswarm/aws-operator/service/awsconfig/v7/key"
 )
 
 func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
-	r.logger.LogCtx(ctx, "level", "debug", "message", "lifecycle resource got executed")
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "looking for the guest cluster cloud formation stack in the AWS API")
+
+	var stackOutputs []*cloudformation.Output
+	{
+		stackOutputs, _, err = r.service.DescribeOutputsAndStatus(key.MainGuestStackName(customObject))
+		if cloudformationservice.IsStackNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the guest cluster cloud formation stack in the AWS API")
+			return nil, nil
+
+		} else if cloudformationservice.IsOutputsNotAccessible(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "the guest cluster cloud formation stack output values are not accessible due to stack state transition")
```

Comment:
For my interest what is `stack state transition`? Does it mean the stack application is still in progress?

Reply:
A stack state transition is when for instance the CF stack is in `CREATE_IN_PROGRESS` and you have to wait for `CREATE_COMPLETE` before being able to access the outputs.